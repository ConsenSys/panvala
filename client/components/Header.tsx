import * as React from 'react';
import { withRouter } from 'next/router';
import Image from './Image';
import Button from './Button';
import RouterLink from './RouterLink';

const Header: React.FunctionComponent = ({ router }: any) => {
  return (
    <header className="mb4">
      <nav>
        <div className="flex justify-between items-center">
          <div className="flex">
            <RouterLink href="/" as="/">
              <Image src="/static/logo-black.svg" alt="panvala logo" />
            </RouterLink>
          </div>
          <div className="flex justify-end">
            <RouterLink href="/" as="/">
              <Button
                active={
                  (router && router.pathname === '/') ||
                  (router && router.asPath && router.asPath.startsWith('/slates'))
                }
              >
                {'Slates'}
              </Button>
            </RouterLink>
            <RouterLink href="/proposals" as="/proposals">
              <Button active={router && router.asPath && router.asPath.startsWith('/proposals')}>
                {'Proposals'}
              </Button>
            </RouterLink>
            <RouterLink href="/ballots" as="/ballots">
              <Button active={router && router.asPath && router.asPath.startsWith('/ballots')}>
                {'Ballots'}
              </Button>
            </RouterLink>
          </div>
        </div>
      </nav>
    </header>
  );
};

export default withRouter(Header);
