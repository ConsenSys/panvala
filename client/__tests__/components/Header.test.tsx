import * as React from 'react';
import { render } from 'react-testing-library';
import Header from '../../components/Header';

describe('Header component', () => {
  test('should render correctly', () => {
    const { container } = render(<Header />);
    expect(container).toMatchSnapshot();
  });
});
