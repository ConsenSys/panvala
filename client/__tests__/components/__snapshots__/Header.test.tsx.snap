// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component should render correctly 1`] = `
<div>
  <header
    class="mb4"
  >
    <nav>
      <div
        class="flex justify-between items-center"
      >
        <div
          class="flex"
        >
          <a
            class="link undefined RouterLink__A-hkpnw2-0 hTfQYN"
          >
            <div
              class="Image__Container-sc-7c9yjn-0 hbTfHQ"
            >
              <div
                class="Image__Wrapper-sc-7c9yjn-1 imCQeS"
              >
                <img
                  alt="panvala logo"
                  class="Image-sc-7c9yjn-2 aZZLn"
                  src="/static/logo-black.svg"
                />
              </div>
            </div>
          </a>
        </div>
        <div
          class="flex justify-end"
        >
          <a
            class="link undefined RouterLink__A-hkpnw2-0 hTfQYN"
          >
            <div
              class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
            >
              <button
                class="Button__StyledButton-sc-1vedn6s-0 lahyEh"
              >
                Slates
              </button>
            </div>
          </a>
          <a
            class="link undefined RouterLink__A-hkpnw2-0 hTfQYN"
          >
            <div
              class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
            >
              <button
                class="Button__StyledButton-sc-1vedn6s-0 lahyEh"
              >
                Proposals
              </button>
            </div>
          </a>
          <a
            class="link undefined RouterLink__A-hkpnw2-0 hTfQYN"
          >
            <div
              class="Button__Wrapper-sc-1vedn6s-2 jHiEij"
            >
              <button
                class="Button__StyledButton-sc-1vedn6s-0 lahyEh"
              >
                Ballots
              </button>
            </div>
          </a>
        </div>
      </div>
    </nav>
  </header>
</div>
`;
